docs/examples/shading/plot_martinez_shade_loss.py

```diff
@@ -0,0 +1,266 @@
+====================================================================
+
+This example demonstrates how to model power loss due to row-to-row shading in
+a PV array comprised of non-monolithic modules.
+"""
+
+# %%
+# This example illustrates how to use the proposed by Martinez et al. [1]_.
+# The model adjusts the output power of a PV module based on the number of
+# shaded *blocks*. A *block* is defined as a group of cells protected by a
+# bypass diode. More information on the *blocks* can be found in the original
+# paper [1]_ and in :py:func:`pvlib.shading.martinez_shade_factor`
+# documentation.
+#
+# The following key functions are used in this example:
+#
+# 1. :py:func:`pvlib.shading.martinez_shade_factor` to calculate the adjustment
+#    factor for the output power.
+# 2. :py:func:`pvlib.shading.shaded_fraction1d` to calculate the fraction of
+#    shaded surface and consequently the number of shaded *blocks* due to
+#    row-to-row shading.
+# 3. :py:func:`pvlib.tracking.singleaxis` to calculate the rotation angle of
+#    the trackers.
+#
+# .. sectionauthor:: Echedey Luis <echelual (at) gmail.com>
+#
+# References
+# ----------
+# .. [1] F. Martínez-Moreno, J. Muñoz, and E. Lorenzo, 'Experimental model
+#    to estimate shading losses on PV arrays', Solar Energy Materials and
+#    Solar Cells, vol. 94, no. 12, pp. 2298-2303, Dec. 2010,
+#    :doi:`10.1016/j.solmat.2010.07.029`.
+#
+# Problem description
+# -------------------
+# Let's consider a PV system with the following characteristics:
+#
+# - Two north-south single-axis tracker with 6 modules each one.
+# - The rows have the same true-tracking tilt angles. Let's consider
+#   true-tracking so shade is significant for this example.
+# - Terrain slope is 7 degrees downward to the east.
+# - Rows' axes are horizontal.
+# - The modules are comprised of multiple cells. We will compare these cases:
+#    - modules with one bypass diode
+#    - modules with three bypass diodes
+#    - half-cut cell modules with three bypass diodes on portrait and landscape
+#
+# Setting up the system
+# ----------------------
+# Let's start by defining the system characteristics, location and the time
+# range for the analysis.
+
+import pvlib
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.dates import ConciseDateFormatter
+
+pitch = 4  # meters
+width = 1.5  # meters
+gcr = width / pitch
+N_modules_per_row = 6
+axis_azimuth = 180  # N-S axis
+axis_tilt = 0  # flat because the axis is perpendicular to the slope
+cross_axis_tilt = -7  # 7 degrees downward to the east
+
+latitude, longitude = 40.2712, -3.7277
+locus = pvlib.location.Location(
+    latitude,
+    longitude,
+    tz="Europe/Madrid",
+    altitude=pvlib.location.lookup_altitude(latitude, longitude),
+)
+
+times = pd.date_range("2001-04-11T03", "2001-04-11T07", periods=24).union(
+    pd.date_range("2001-04-11T16", "2001-04-11T20", periods=24)
+)
```

Comment:
Why exclude the middle of the day?

Reply:
To save some computation time while preserving the output. It doesn't get shaded during those hours. I ridiculously microoptimize too much. Do you think its impact on readability is too significant? I don't have a problem with changing it.

Reply:
I think the benefit (save a few ms at runtime) is not worth the cost (lose ~10s at "readtime") :)